File: Java/remove-duplicates-from-sorted-array-ii.java
//  Remove Duplicates from Sorted Array II

// Time O(N), Space O(1)
class Solution {
    public int removeDuplicates(int[] nums) {
        int len = 2;
        if (nums.length <= 2)
            return nums.length;
        for (int i=2; i<nums.length; i++)
            if (nums[i] != nums[len-2])
                nums[len ++] = nums[i];
        return len;
    }
}

File: leetcode/remove-duplicates-from-sorted-array-ii.cc
//  Remove Duplicates from Sorted Array II

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.size() < 3)    return nums.size();
        int res = 2;
        for (int i=2; i<nums.size(); i++) 
            if (nums[i] != nums[res-2])
                nums[res ++] = nums[i];
        return res;
    }
};